nestkernel/source_table.cpp

```diff
@@ -357,9 +362,26 @@ nest::SourceTable::get_next_target_data( const thread tid,
         next_target_data.set_is_primary( true );
         // we store the thread index of the source table, not our own tid!
         TargetDataFields& target_fields = next_target_data.target_data;
-        target_fields.set_tid( current_position.tid );
         target_fields.set_syn_id( current_position.syn_id );
-        target_fields.set_lcid( current_position.lcid );
+        if ( current_source.is_compressed() )
+        {
+          target_fields.set_tid( MAX_TID );  // use MAX_TID as marker for compressed spikes
+          auto it_idx = compressed_spike_data_map_[ current_position.tid ].find( current_source.get_gid() );
```

Comment:
Use of `auto` to be avoided?

Reply:
@7eia No, use of C++11 constructs is encouraged in new or refactored code.

Reply:
@heplesser Thank you for clarifying.